### PR TITLE
[JENKINS-66969] Fix ongoing first build from disappearing from build history

### PR DESCRIPTION
**Pull request by janfaracik**
See [JENKINS-66969](https://issues.jenkins.io/browse/JENKINS-66969).

---

This PR also does a bit of cleanup/some general bug fixes to the build history widget, if y'all want me to split out the bug fix by itself happy to do so but this PR does provide value apart from that bug fix -

* Cleans up some of the JS selectors
* Entries that are loaded in dynamically are now added to `tbody` rather than the table so styling is set correctly without needing to reload
* Fixed case where the "No builds" banner could show even when there were builds visible

---

### Proposed changelog entries

* Show ongoing first build in build history (regression in 2.314).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by MarkEWaite — COMMENTED** [review]
In order to backport the change to the next LTS baseline, 2.319.1, I think that we'll need it split into two pull requests.

That's based on the assumption that the other improvements may need more time to be evaluated in weekly and may be focused on the 2.320 code base rather than on the code that is in 2.319.

**Review by timja — COMMENTED** [review]
Can you minimalise the diff please, none of the semi colon or variable type changes

**Comment by janfaracik** [comment]
> In order to backport the change to the next LTS baseline, 2.319.1, I think that we'll need it split into two pull requests.
> 
> That's based on the assumption that the other improvements may need more time to be evaluated in weekly and may be focused on the 2.320 code base rather than on the code that is in 2.319.

Sure thing, should I branch off of https://github.com/jenkinsci/jenkins/tree/jenkins-2.319 ?

**Comment by janfaracik** [comment]
> Can you minimalise the diff please, none of the semi colon or variable type changes

Done :)

**Comment by MarkEWaite** [comment]
> > In order to backport the change to the next LTS baseline, 2.319.1, I think that we'll need it split into two pull requests.
> > That's based on the assumption that the other improvements may need more time to be evaluated in weekly and may be focused on the 2.320 code base rather than on the code that is in 2.319.
> 
> Sure thing, should I branch off of https://github.com/jenkinsci/jenkins/tree/jenkins-2.319 ?

I think that will be best since that will make it easier to include in the backports for 2.319.1

File: core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly
Comment (timja) [inline core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly:87]: why not just (untested):
(doesn't matter if there's an empty style tag)
```suggestion
          <div id="no-builds" class="jenkins-pane__information" style="${page.runs.isEmpty() ? "" : "display: none;"}">
            ${%No builds}
          </div>
```

File: war/src/main/js/filter-build-history.js
Comment (timja) [inline war/src/main/js/filter-build-history.js:31]: seems a bit fragile =/

**Review by timja — APPROVED** [review]
probably ok, tested and it works.

seems small enough for backporting, any other thoughts?

File: core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly:87]: This makes a ton more sense - not too sure what was going through my head 🙈 Thanks :)

File: core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly
Comment (timja) [inline core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly:26]:
```suggestion
<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:i="jelly:fmt">
```

File: core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly
Comment (timja) [inline core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly:82]: quoting is off here, (suggestion was untested)

**Review by fqueiruga — APPROVED** [review]

File: core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly]: Fixed 👍

File: war/src/main/js/filter-build-history.js
Comment (janfaracik) [reply war/src/main/js/filter-build-history.js:31]: Agreed, the widget itself is something I want to take another look at in the future so I'll likely update this in the future.

**Review by timja — APPROVED** [review]
re-tested good now

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
